presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -4119,22 +4137,31 @@ public void testPredicatePushDownToTableScan()
         }
     }
 
-    @Test
-    public void testParquetTimestampPredicatePushdown()
+    @DataProvider
+    public Object[][] timestampPrecisionAndValues()
+    {
+        // TODO: revisit values once we handle write path
+        return new Object[][] {{3, "2012-10-31 01:00:08.123"}, {6, "2012-10-31 01:00:08.123000"}, {9, "2012-10-31 01:00:08.123000000"}};
+    }
+
+    @Test(dataProvider = "timestampPrecisionAndValues")
+    public void testParquetTimestampPredicatePushdown(int precision, String value)
```

Comment:
You put the value into SQL (as literal's content). This is OK if and only if you know it's TIMESTAMP literal and you know the contents do not contain `'`.
This slightly blurs responsibility. LDT can hold any value you want and allows you to put _value -> SQL literal_ conversion in single place.
This also allows you to add tests with `value - 1  nano < t AND t value + 1 nano` (because LDT lets you add to the value).

of course, i am aware of some readability drawbacks, so you may want to actually compare the code with and without LDT